python/argo_client/netstring.py

```diff
@@ -22,20 +28,26 @@ def decode(netstring : bytes) -> Tuple[str, bytes]:
 
     """
 
-    i = 0
-    length_bytes = bytearray(b'')
-    while chr(netstring[i]).isdigit():
-        length_bytes.append(netstring[i])
-        i += 1
-    if chr(netstring[i]).encode() != b':':
-        raise ValueError("Malformed netstring, missing :")
-    length = int(length_bytes.decode())
-    i += 1
-    out = bytearray(b'')
-    for j in range(0, length):
-        out.append(netstring[i])
-        i += 1
-    if chr(netstring[i]).encode() != b',':
-        raise ValueError("Malformed netstring, missing ,")
-    i += 1
-    return (out.decode(), netstring[i:])
+    colon = netstring.find(b':')
+    if colon == -1 and len(netstring) >= 10 or colon >= 10:
+        # cut things off at about a gigabyte
+        raise InvalidNetstring("message length too long")
```

Comment:
I think I'm missing something: how does this code check if the netstring is over a gigabyte? (The number `10` would suggest a much smaller amount.)

Reply:
a gigabyte is 1073741824 which is ten digits long. if we're getting more than 10 digits then something is wrong and I wanted to detect that early. I wasn't really intending for this to be a meaningful size restriction. I can do that if it's important

Reply:
Ah, I thought you meant gigabytes in the sense of the _length_ of the netstring, but you meant gigabytes in the _value_ of the netstring. That clarifies my confusion (although it would be worth adding a similar clarification in the comment as well).